install/installer/cluster_dev.py

```diff
@@ -372,6 +507,455 @@ def choose_git_provider(cli_arg, repo):
+    Returns:
+        git_token string
+    """
+
+    # Required env variables from host: GITHUB_TOKEN
+    if provider == 'Github':
+        if 'GITHUB_TOKEN' in os.environ:
+            git_token = os.environ['GITHUB_TOKEN']
+            print('Use GITHUB_TOKEN from env')
+        else:
+            git_token = ask_user('github_token')
+
+    return git_token
+
+
+@typechecked
+def get_repo_name_from_url(url: str) -> str:
+    """
+    Get git repository name from git remote url
+    """
+    last_slash_index = url.rfind("/")
+    last_suffix_index = url.rfind(".git")
+
+    if last_slash_index < 0 or last_suffix_index < 0:
+        _exit_(f'ERROR: Check `git remote -v`. Badly formatted origin: {url}')
+
+    return url[last_slash_index + 1:last_suffix_index]
+
+
+@typechecked
+def get_repo_owner_from_url(url: str) -> str:
+    """
+    Get git repository owner from git remote url
+    """
+    last_slash_index = url.rfind("/")
+
+    # IF HTTPS: https://github.com/shalb/cluster.dev.git
+    prefix_index = url.find("/", 8)
+    # IF SSH: git@github.com:shalb/cluster.dev.git
+    if prefix_index == last_slash_index:
+        prefix_index = url.find(":")
+
+    if last_slash_index < 0 or prefix_index < 0:
+        _exit_(f'ERROR: Check `git remote -v`. Badly formatted origin: {url}')
+
+    return url[prefix_index + 1:last_slash_index]
+
+
+@typechecked
+def encrypt(public_key: str, secret_value: str) -> str:
+    """Encrypt a Unicode string using the public key."""
+    public_key = public.PublicKey(public_key.encode("utf-8"), encoding.Base64Encoder())
+    sealed_box = public.SealedBox(public_key)
+    encrypted = sealed_box.encrypt(secret_value.encode("utf-8"))
+    return b64encode(encrypted).decode("utf-8")
+
+
+@typechecked
+def create_github_secrets(creds: dict, cloud: str, repo: Repo, git_token: str):
+    """
+    Create Github repo secrets for specified Cloud.
+
+    Args:
+        creds (dict(str)): Programmatic access to cloud
+            { 'key': str, 'secret': str, ... }
+        cloud (str): Cloud name
+        repo (obj): git.Repo object
+        git_token(str): Git Provider token
+    """
+
+    gh_api = GitHub(token=git_token)
+
+    remote = repo.remotes.origin.url
+    owner = get_repo_owner_from_url(remote)
+    repo_name = get_repo_name_from_url(remote)
+
+    # Get public key for encryption
+    # https://developer.github.com/v3/actions/secrets/#get-a-repository-public-key
+    for i in range(3, -1, -1):
+        if i == 0:
+            _exit_('ERROR: Can\'t access Github. Please, try again later')
+        try:
+            status, public_key = getattr(
+                getattr(
+                    getattr(
+                        gh_api.repos, owner,
+                    ), repo_name,
+                ).actions.secrets, 'public-key',
+            ).get()
+            break
+        except TimeoutError:
+            print(f'Can\'t access Github. Timeout error. Attempts left: {i}')
```

Comment:
Say no to more than one instruction in try-block:
```suggestion
        except TimeoutError:
            print(f'Can\'t access Github. Timeout error. Attempts left: {i}')
        else:
            break
```

Reply:
Hmm, `try/else`, I will know.